[Bug] No matching toolchains found for requested specification: {languageVersion=11, vendor=any, implementation=vendor-specific}.

Reported by @criticalAY 

```
* What went wrong:
Could not determine the dependencies of task ':lint-rules:ktlintKotlinScriptFormat'.
> Could not resolve all dependencies for configuration ':lint-rules:ktlint'.
> Failed to calculate the value of task ':lint-rules:compileJava' property 'javaCompiler'.
> No matching toolchains found for requested specification: {languageVersion=11, vendor=any, implementation=vendor-specific}.
> No locally installed toolchains match (see https://docs.gradle.org/8.0.1/userguide/toolchains.html#sec:auto_detection) and toolchain download repositories have not been configured (see https://docs.gradle.org/8.0.1/userguide/toolchains.html#sub:download_repositories). Does anyone have idea about this error I am not able to commit my changes
```

Probable cause: 625e6c71cb9b1bc7f39c4b4391c478bee59b6a4b

Creating issue for documentation/tracking

Additional bug when moved to JDK11

```
* What went wrong:
Execution failed for task ':AnkiDroid:runKtlintFormatOverMainSourceSet'.
> A failure occurred while executing org.jlleitschuh.gradle.ktlint.worker.KtLintWorkAction
> java.lang.ExceptionInInitializerError (no error message)
```

The second error was
```
* What went wrong:
Execution failed for task ':AnkiDroid:runKtlintFormatOverMainSourceSet'.
> A failure occurred while executing org.jlleitschuh.gradle.ktlint.worker.KtLintWorkAction
   > java.lang.ExceptionInInitializerError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 13s
```
There was no java error in this one but the ktlint command fails for some reason though when I ran ktlint over main branch it was successful

--- a/lint-rules/build.gradle
+++ b/lint-rules/build.gradle
@@ -5,8 +5,15 @@ repositories {
     mavenCentral()
 }
 
-kotlin {
-    jvmToolchain(11)
+tasks.withType(JavaCompile).configureEach {
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
+}
+
+tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
+    kotlinOptions {
+        jvmTarget = JavaVersion.VERSION_11 // starting with AGP 7.4.0 we need to target JVM 11 bytecode
+    }
 }
 
 dependencies {

